frontend/toolbar/mainToolbar.component.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+(function () {
+    const app = angular.module('app');
+
+    app.component("mainToolbar", {
+        templateUrl: 'app/toolbar/main_toolbar_mobile.html',
+        controller: 'MainToolbarController',
+        controllerAs: 'mainToolbarCtrl',
+        bindings: {
+            title: '=',
+            toolbarMenuItems: '=',
```

Comment:
these toolbbarMenuItems and toolbarGeneralOptions couldn't be models? just to be easier to find out whats the structure of the object needed?

Reply:
They could, but I would have to pass their field completely from the controller to the constructor once the fields, despite they have the same signature, changes a lot from events to notifications for example. Even though, should I try it?

Reply:
I will put as an issue, for later.